web/app.js

```diff
@@ -2674,18 +2702,66 @@ function webViewerWheel(evt) {
 
 function webViewerTouchStart(evt) {
   if (evt.touches.length > 1) {
-    // Disable touch-based zooming, because the entire UI bits gets zoomed and
-    // that doesn't look great. If we do want to have a good touch-based
-    // zooming experience, we need to implement smooth zoom capability (probably
-    // using a CSS transform for faster visual response, followed by async
-    // re-rendering at the final zoom level) and do gesture detection on the
-    // touchmove events to drive it. Or if we want to settle for a less good
-    // experience we can make the touchmove events drive the existing step-zoom
-    // behaviour that the ctrl+mousewheel path takes.
     evt.preventDefault();
+
+    if (evt.touches.length !== 2) {
+      PDFViewerApplication._touchInfo = null;
+      return;
+    }
+
+    const [touch0, touch1] = evt.touches;
+    PDFViewerApplication._touchInfo = {
+      distance:
+        Math.hypot(touch0.pageX - touch1.pageX, touch0.pageY - touch1.pageY) ||
+        1,
+    };
```

Comment:
If this is only going to store *a single* property, at least for now, can't we skip the object and just do the following instead?
```suggestion
    PDFViewerApplication._touchDistance =
        Math.hypot(touch0.pageX - touch1.pageX, touch0.pageY - touch1.pageY) ||
        1;
```

Although, this would probably require that you *explicitly* check `PDFViewerApplication._touchDistance !== null` in the functions below.